dashboard/app/controllers/programming_classes_controller.rb

```diff
@@ -30,9 +30,20 @@ def update
       render :not_found
       return
     end
-    @programming_class.assign_attributes(programming_class_params.except(:category_key))
+    @programming_class.assign_attributes(programming_class_params.except(:category_key, :methods))
     programming_environment_category = @programming_class.programming_environment.categories.find_by_key(programming_class_params[:category_key])
     @programming_class.programming_environment_category_id = programming_environment_category&.id
+    if programming_class_params[:methods]
+      @programming_class.programming_methods = programming_class_params[:methods].each_with_index.map do |method_params, i|
+        if method_params['id']
+          method = ProgrammingMethod.find(method_params['id'])
+          method.update!(method_params.merge(position: i))
+          method
+        else
+          ProgrammingMethod.create!(method_params.merge(programming_class_id: @programming_class.id, position: 1))
```

Comment:
Is the position always going to be 1 here, would you ever be creating more than one?

Reply:
omg good catch. This should be `i`